wsd/COOLWSD.cpp

```diff
@@ -1003,6 +1010,254 @@ COOLWSD::~COOLWSD()
+
+        sslRemoteParams.verificationMode = (sslRemoteParams.caLocation.empty() ? Poco::Net::Context::VERIFY_NONE : Poco::Net::Context::VERIFY_STRICT);
+        sslRemoteParams.loadDefaultCAs = true;
+        Poco::Net::Context::Ptr sslClientContext = new Poco::Net::Context(Poco::Net::Context::CLIENT_USE, sslRemoteParams);
+
+        Poco::Net::SSLManager::instance().initializeClient(consoleClientHandler, invalidClientCertHandler, sslClientContext);
+
+        startThread();
+    }
+
+    void pollingThread()
+    {
+        while (!isStop() && !SigUtil::getTerminationFlag() && !SigUtil::getShutdownRequestFlag())
+        {
+            try
+            {
+                std::shared_ptr<http::Session> httpSession(StorageBase::getHttpSession(remoteServerURI));
+                http::Request request(remoteServerURI.getPathAndQuery());
+                if (!eTagValue.empty())
+                {
+                    request.set("If-None-Match", eTagValue);
+                }
+                request.set("User-Agent", WOPI_AGENT_STRING);
+
+                const std::shared_ptr<const http::Response> httpResponse
+                    = httpSession->syncRequest(request);
+
+                unsigned int statusCode = httpResponse->statusLine().statusCode();
+                if (statusCode == Poco::Net::HTTPResponse::HTTP_NOT_MODIFIED)
+                {
+                    throw Poco::Exception("Response has status code: " + std::to_string(statusCode) + ", JSON has not been changed");
+                }
+                else if (statusCode != Poco::Net::HTTPResponse::HTTP_OK)
+                {
+                    throw Poco::Exception("Response has status code: " + std::to_string(statusCode));
+                }
+
+                eTagValue = httpResponse->get("ETag");
+
+                std::string body = httpResponse->getBody();
+
+                Poco::JSON::Object::Ptr remoteJson;
+                if (JsonUtil::parseJSON(body, remoteJson))
+                {
+                    std::string kind;
+                    JsonUtil::findJSONValue(remoteJson, "kind", kind);
+                    if (kind != "configuration")
+                    {
+                        throw Poco::JSON::JSONException("Make sure that JSON is for remote configuration, check the value of kind property");
+                    }
+                    applyNewConfig(remoteJson);
+                }
+                else
+                {
+                    throw Poco::JSON::JSONException("Could not parse the JSON");
+                }
+            }
+            catch(Poco::Exception& exc)
+            {
+                LOG_ERR(exc.displayText());
+            }
+            poll(std::chrono::seconds(60));
+        }
+    }
+
+    void applyNewConfig(Poco::JSON::Object::Ptr remoteJson)
+    {
+        std::map<std::string, std::string> newAppConfig;
+
+        fetchWopiHostPatterns(newAppConfig, remoteJson);
+
+        #ifdef ENABLE_FEATURE_LOCK
```

Comment:
We don't use the convention of indenting preprocessor lines anywhere else in the code, please don't start using it here. Consistency is important. (And then no extra indentation of the lines inside the ifdef block either.)